parse/ValueRefParser.h

```diff
@@ -188,6 +189,21 @@ namespace parse { namespace detail {
             |   unwrapped_bound_variable [ _val = _1 ]
             ;
     }
+
+    template <typename T>
+    void open_and_register_as_string(std::string& nameref, ::parse::detail::MovableEnvelope<T>& obj, bool& pass)
+    {
+      if (obj.IsEmptiedEnvelope()) {
+	ErrorLogger() <<
+	  "The parser attempted to extract the unique_ptr from a MovableEnvelope more than once - while looking at a valueref envelope for use in ValueRef registration ";
+	pass = false;
+	return;
+      }
+      ::RegisterValueRef<T>(nameref, std::move(obj.OpenEnvelope(pass)));
```

Comment:
I think the `move` is unnecessary as the return value of `OpenEnvelope` should already be an rvalue reference to `unique_ptr`

Reply:
the name gets reused (wrapped in qi laziness _2), so i rather make a copy when calling the global RegisterValueRef